doc: Crypto samples doc (#4779)
nrfconnect/sdk-nrf: samples/crypto/ecdh/README.rst

   :local:
   :depth: 2

> The ECDH sample shows how to perform an Elliptic-curve Diffie–Hellman key exchange to allow two parties to obtain a common secret.

Reviewer: Please change "common secret" to "shared secret" as this is  statement that is standardized phrase in Diffie-Hellman key exchange .

> The ECDH sample shows how to perform an Elliptic-curve Diffie–Hellman key exchange to allow two parties to obtain a common secret.

Reviewer 2: ```suggestion
The ECDH sample shows how to perform an Elliptic-curve Diffie–Hellman key exchange to allow two parties to obtain a shared secret.
```

Overview
********

The sample follows these steps to perform an ECDH key exchange:

1. First, the sample performs initialization:

   a. The Platform Security Architecture (PSA) API is initialized.

>    #. Two random ECDH key pairs are generated and imported into the PSA crypto keystore.

Reviewer: ECDH key pairs needs to be changed ECC key pairs. There is nothing that makes these key pairs specific to an ECDH exchange.

Bonus points: Explain ECC  as "Elliptic Curve Cryptography" as this is possibly the first time that term is used (and the acronym is probably not that well known). PSA uses this acronym for their APIs as well...

@Vge0rge : This probably needs to be fixed in the code if the term ECDH keypair is used

Reviewer 2: I used the term ECDH because the algorithm that we set in the psa_apis is ECDH. But I understand that if we want to be technically correct probably this needs to be changed in the code as well. The function names comments as well. So my opinion is to leave it like that for now and just create a PR after the release to fix both the code and documentation.

Reviewer: That is acceptable for me, @ioannisg?

Author: Changed to ECC as suggested

>    #. Two random ECDH key pairs are generated and imported into the PSA crypto keystore.

Reviewer 2: ```suggestion
   #. Two random Elliptic Curve Cryptography(ECC) key pairs are generated and imported into the PSA crypto keystore.
```

#. A sample plaintext is signed:

>    a. Signing is performed.

Reviewer: ECDH does not include signing. Is this a copy-paste from ECDSA example?

The operations is supposed to be a key exchange (sending the public key to the other end) and then calculating a shared secret on both ends...

Author: @Vge0rge  ?

Reviewer 2: That is correct, the signing part needs to be removed completely.

   #. The key pair is removed from the PSA keystore.

#. Then, ECDH key exchange is performed using the generated key pairs.

#. Afterwards, the sample performs cleanup:

>    a. The ECDH key pairs are removed from the PSA crypto keystore.

Reviewer 2: ```suggestion
   a. The key pairs are removed from the PSA crypto keystore.
```
